nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -7,148 +7,276 @@
   ...
 }:
 
-with lib;
-
 let
   cfg = config.virtualisation.xen;
+
+  xenBootBuilderPath = "${
+    pkgs.writeShellApplication {
```

Comment:
I would like to see this package factored out into its own variable, so that you don’t have to keep the huge `${…}` in your mental context stack to make sense of the path at the end. (Maybe you could use `lib.getExe`, even?)